common/main/java/com/couchbase/lite/internal/core/C4CollectionDocObserver.java

```diff
@@ -21,11 +21,11 @@ public interface NativeImpl {
     //-------------------------------------------------------------------------
 
     @NonNull
-    private static final NativeImpl NATIVE_IMPL = new NativeC4CollectionDocObserver();
+    @VisibleForTesting
```

Comment:
I don't think there is any need for this.  You should never need to assign `nativeImpl` unless it is used in the `callback` method.  I don't think it is.